mantidproject/mantidimaging
# Refactor registration to make it much more generic

Contains #136 and requires it to be merged first.

Cleans up the registration code with the aim of making it more flexible when searching for and operating on child packages/modules.

Required for the new filter GUI structure (#138).

To test:
- Open the GUI, see that all the filters that were previously listed still are
- Use `mantidimaging --help`, see that all the filter options that were there previously are still there
